enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,321 @@
+# placement extensible scheduling
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposed work provides an API to represent a scalar value of the managed cluster, to support placement extensible scheduling.
+
+## Motivation
+
+When implementing placement resource based scheduling, we find in some cases, prioritizer needs extra data (more than the defualt value provided by `ManagedCluster` and `ManagedClusterInfo`) to calculate the score of the managed cluster.
+
+For example, prioritizer ResourceRatioCPU depends on knowing the real-time CPU usage, which is not provided by `ManagedCluster`.
+So we want a more extensible way to support scheduling based on customized value.
+
+### User Stories
+
+#### Story 1: The placement controller could prioritize clusters by data pushed from each managed cluster.
+  - As an end user, can define prioritizer name `customized-resourceratiocpu` in placment spec.prioritizerPolicy.configurations to say he wants to schedule based on customized value resourceratiocpu.
+  - On each managed cluster, there would be a customized agent to sum up the cpu usage of the cluster. Then update the value into the ManagedClusterScalar CR with name `customized-resourceratiocpu`.
+  - On the hub, the placement prioritizer will read value from ManagedClusterScalar `customized-resourceratiocpu` and score based on it. 
+
+#### Story 2: The placement controller could prioritize clusters by metrics collected on hub.
+  - As an end user, can define prioritizer name `customized-resourceratiocpu` in placment spec.prioritizerPolicy.configurations to say he wants to schedule based on customized value resourceratiocpu.
+  - On the hub, there would be a customized agent to read cpu usage from thanos. It will generate a value for each cluster and update it into ManagedClusterScalar CR with name `customized-resourceratiocpu`.
+  - On the hub, the placement prioritizer will read value from ManagedClusterScalar `customized-resourceratiocpu` and score based on it.
+
+#### Story 3: The placement controller could prioritize clusters for disaster recovery.
```

Comment:
This story needs to be rephrased a bit. I think briefly:

1. a user has two clusters, a primary and backup cluster on which workload storage is configured so storage are synced from primary cluster to backup.
2. user want to deploy workloads on the primary cluster.
3. when primary cluster is down, the workload should be automatically switched to the backup cluster.